server/src/main/java/org/candlepin/pinsetter/tasks/PopulateHostedDBTask.java

```diff
@@ -102,22 +105,36 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
             log.info("Importing data for all {} known products...", productIds.size());
         }
 
+        int confBatchSize = config.getInt(ConfigProperties.POPULATE_HOSTED_DB_JOB_PROD_LOOKUP_BATCH_SIZE);
         while (productIds.size() > 0) {
-            for (Product product : this.productService.getProductsByIds(productIds)) {
-                if (product != null) {
-                    log.info("Storing product: {}", product);
-
-                    dependentProducts.addAll(product.getDependentProductIds());
-
-                    for (ProductContent pcontent : product.getProductContent()) {
-                        log.info("  Storing product content: {}", pcontent.getContent());
-                        this.contentCurator.createOrUpdate(pcontent.getContent());
-                        ++ccount;
+            log.info("Fetching and processing {} products from the adapter.", productIds.size());
+            // Batch fetch and process the product ids to avoid long request
+            // times from the adapter causing c3p0 to drop idle connections.
+            int adapterLookupCount = 0;
+            for (List<String> productIdBatch : Iterables.partition(productIds, confBatchSize)) {
+                // Nested loop to allow DB activity.
+                adapterLookupCount++;
+                log.info("Batch fetching products from adapter: {} of {} batches of {}.", adapterLookupCount,
+                    (int) Math.ceil((double)productIds.size() / (double) confBatchSize), confBatchSize);
+                for (Product product :
+                    this.productService.getProductsByIds(new HashSet<String>(productIdBatch))) {
```

Comment:
We don't need to drop the block into a HashSet here, since the service takes a generic collection. Even though we know there are inefficiencies on their end, doing an extra copy on our end for a list that is likely already de-duplicated probably isn't worth the overhead (nor the extra line of code from having to wrap a long-line).